### PR TITLE
feat(test): test notifications

### LINKED ISSUE #34: [Test] Test notifications

### PR DESCRIPTION
**Pull request by heafalan**
Fixes #34

### CODE REVIEW

File: packages/test/jest.test.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ const hrp = require("http-request-plus").default;
 const jrp = require("json-rpc-protocol");
 const keyBy = require("lodash/keyBy");
 
+const websocket = new WebSocket("ws://localhost:4000");
```
Comment (julien-f) [inline packages/test/jest.test.js]: Move into `beforeAll`.

File: packages/test/jest.test.js [hunk 2]
```diff
@@ -27,12 +29,49 @@ function compareTimestamps(ts1, ts2) {
...
 }
 
-let entries, id1, id2, id3;
+let entries, id1, id2, id3, message;
+
+beforeAll(() => {
+  websocket.onmessage = event => {
+    message = jrp.parse(event.data);
+  };
+});
```
Comment (julien-f) [inline packages/test/jest.test.js:48]: You should wait for the connection to be open.

File: packages/test/jest.test.js [hunk 3]
```diff
@@ -27,12 +29,49 @@ function compareTimestamps(ts1, ts2) {
...
   id1 = await call("createEntry", { name: "name1", content: "content1" });
+  expect(message.type).toBe("notification");
+  expect(message.method).toBe("createEntry");
+  expect(message.params.entry).toEqual({
+    id: id1,
+    name: "name1",
+    content: "content1",
+    created: expect.any(Number),
+    updated: expect.any(Number),
+  });
```
Comment (julien-f) [inline packages/test/jest.test.js:73]: You could probably simplify the test by comparing with the result of `listEntries`.

File: packages/test/jest.test.js [hunk 4]
```diff
@@ -158,8 +226,19 @@ test("Error: method not found", async () => {
 
 test("delete entry", async () => {
   await call("deleteEntry", { id: id1 });
+  expect(message.type).toBe("notification");
+  expect(message.method).toBe("deleteEntry");
+  expect(message.params.id).toEqual(id1);
```
Comment (julien-f) [inline packages/test/jest.test.js]: Use `toEqual` to simplify the tests.

File: packages/test/jest.test.js [hunk 5]
```diff
@@ -27,36 +27,78 @@ function compareTimestamps(ts1, ts2) {
...
+
+let websocket;
+
+beforeAll(async () => {
+  websocket = new WebSocket("ws://localhost:4000");
+
+  await new Promise(resolve => {
+    websocket.onopen = resolve;
+  });
+});
```
Comment (julien-f) [inline packages/test/jest.test.js:48]:
```js
let entries, id1, id2, id3, waitMessage;

beforeAll(() => {
  let resolveWaitMessage = Function.prototype;
  waitMessage = () =>
    new Promise(resolve => {
      resolveWaitMessage = resolve;
    });

  websocket.onmessage = event => {
    resolveWaitMessage(jrp.parse(event.data));
  };
});
```

File: packages/test/jest.test.js [hunk 6]
```diff
@@ -27,36 +27,78 @@ function compareTimestamps(ts1, ts2) {
...
+  expect(message1.params.entry).toEqual({
+    id: id1,
+    name: "name1",
+    content: "content1",
+    created: expect.any(Number),
+    updated: expect.any(Number),
+  });
+
+  setUpMessage();
   id2 = await call("createEntry", { content: "content2" });
```
Comment (julien-f) [inline packages/test/jest.test.js]:
```js
  let message;

  [message, id1] = await Promise.all([
    waitMessage(),
    call("createEntry", { name: "name1", content: "content1" }),
  ]);
```

File: packages/test/jest.test.js [hunk 7]
```diff
@@ -27,36 +27,81 @@ function compareTimestamps(ts1, ts2) {
...
+  ]);
+  expect(message1.type).toEqual("notification");
+  expect(message1.method).toEqual("createEntry");
+  expect(message1.params.entry).toEqual({
+    id: id1,
+    name: "name1",
+    content: "content1",
+    created: expect.any(Number),
+    updated: expect.any(Number),
+  });
```
Comment (julien-f) [inline packages/test/jest.test.js:73]: Why not using `toEqual` on the whole message?

File: packages/test/jest.test.js [hunk 8]
```diff
@@ -27,36 +27,81 @@ function compareTimestamps(ts1, ts2) {
...
 test("create entry", async () => {
-  id1 = await call("createEntry", { name: "name1", content: "content1" });
-  id2 = await call("createEntry", { content: "content2" });
-  id3 = await call("createEntry", { name: "name3" });
+  let message1;
+  [message1, id1] = await Promise.all([
+    waitMessage(),
+    call("createEntry", { name: "name1", content: "content1" }),
+  ]);
+  expect(message1.type).toEqual("notification");
```
Comment (julien-f) [inline packages/test/jest.test.js]: Use `toEqual` for deep comparisons and `toBe` for scalars.

File: packages/test/jest.test.js [hunk 9]
```diff
@@ -27,36 +27,96 @@ function compareTimestamps(ts1, ts2) {
...
-  id2 = await call("createEntry", { content: "content2" });
-  id3 = await call("createEntry", { name: "name3" });
+  let message1;
+  [message1, id1] = await Promise.all([
+    waitMessage(),
+    call("createEntry", { name: "name1", content: "content1" }),
+  ]);
+  expect(message1.type).toBe("notification");
+  expect(message1).toEqual({
+    jsonrpc: expect.any(String),
```
Comment (julien-f) [inline packages/test/jest.test.js]: Why `any(String)`?

Comment (heafalan) [reply packages/test/jest.test.js]: I thought the version of json-rpc protocol could change, but it MUST stay the same ('2.0'). I change this!

**Pull request merged**